### PR TITLE
chore(deps): bump and use workspace dependencies

### LINKED ISSUE #1963: Support workspace dependencies
Cargo supports [inheriting dependencies from the workspace](https://doc.rust-lang.org/stable/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace) since v1.64, which is now our MSRV. We should consider supporting this to pin common dependencies to the same version and features, and make them more maintainable.

I'm aware we have dependabot but it looks like it only updates one crate's manifest and then the root lock file, creating different pinned versions, see examples.

I'm not sure these different versions have any effect when importing the main crate `ethers`, but it might when importing the individual crates, not sure since they might still use the version in the lock file (?)

I'm happy to implement this if it's something we want

## Examples

`syn`:

syn

`serde_json`:

serde_json

```bash
#!/bin/bash
crates=(ethers-addressbook ethers-contract ethers-contract/ethers-contract-abigen ethers-contract/ethers-contract-derive ethers-core ethers-core/ethers-derive-eip712 ethers-etherscan ethers-middleware ethers-providers ethers-signers ethers-solc)
for c in "${crates[@]}"
do
    res="$(cat "$c/Cargo.toml" | grep "$@")"
    [ "$?" -eq 0 ] && printf "$c:\n$res\n\n"
done
```

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Closes #1963

## Solution

See commit log

## PR Checklist

-   [ ] Added Tests
-   [ ] Added Documentation
-   [ ] Updated the changelog
-   [ ] Breaking changes

### CODE REVIEW

**Comment by gakonst** [comment]
sorry need to bump versions to 2.0 everywhere

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -147,9 +165,6 @@ jobs:
             - uses: Swatinem/rust-cache@v2
             - name: Check
               run: cargo check --workspace --target wasm32-unknown-unknown
-            # TODO: [#2191](https://github.com/gakonst/ethers-rs/issues/2191)
```
Comment (gakonst) [inline .github/workflows/ci.yml:150]: is this no longer a problem?

File: ethers-providers/src/rpc/transports/retry.rs [hunk 2]
```diff
@@ -338,9 +335,7 @@ where
                 trace!("retrying and backing off for {:?}", next_backoff);
 
                 #[cfg(target_arch = "wasm32")]
-                wasm_timer::Delay::new(next_backoff)
-                    .await
-                    .map_err(|_| RetryClientError::TimerError)?;
+                futures_timer::Delay::new(next_backoff).await;
```
Comment (gakonst) [inline ethers-providers/src/rpc/transports/retry.rs:338]: is this correct? if so, should we just use it in both cases?

File: ethers-solc/src/compile/mod.rs [hunk 3]
```diff
@@ -70,7 +70,7 @@ pub(crate) fn take_solc_installer_lock() -> std::sync::MutexGuard<'static, ()> {
 /// A list of upstream Solc releases, used to check which version
 /// we should download.
 /// The boolean value marks whether there was an error accessing the release list
-#[cfg(all(feature = "svm-solc"))]
+#[cfg(all(feature = "svm-solc", not(target_arch = "wasm32")))]
```
Comment (gakonst) [inline ethers-solc/src/compile/mod.rs:73]: Does anything in ethers-solc work inside of WASM, given it's calling the binary? Should we just make the crate not compile with ethers-rs if wasm is not configured?

File: examples/anvil/Cargo.toml [hunk 4]
```diff
@@ -1,10 +1,14 @@
 [package]
```
Comment (gakonst) [inline examples/anvil/Cargo.toml:1]: Can we make `cargo release` not go over the examples crates if possible? Right now it thinks all the examples should be published

File: ethers/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,89 @@
+[package]
+name = "ethers"
+authors = ["Georgios Konstantopoulos <me@gakonst.com>"]
+readme = "../README.md"
+description = "A complete Ethereum and Celo Rust library"
```
Comment (gakonst) [inline ethers/Cargo.toml:5]: Are we OK with pulling ethers back down a directory? I initially had pulled it up to surface the examples, but seems fine now given we have separate examples crates

**Review by gakonst — APPROVED** [review]
fine w me - want +1 from @mattsse

File: examples/anvil/Cargo.toml (hunk 4 above)
Comment (DaniPopes) [reply examples/anvil/Cargo.toml:1]: doesn't `publish = false` do the trick? What's the command you run to release

File: .github/workflows/ci.yml (hunk 1 above)
Comment (DaniPopes) [reply .github/workflows/ci.yml:150]: hmm i forget why i removed that, it still is with the features

File: ethers/Cargo.toml (hunk 5 above)
Comment (DaniPopes) [reply ethers/Cargo.toml:5]: I think so, also this way it doesn't pull in every folder like `.github` when packaging the `ethers` crate for release

**Review by gakonst — APPROVED** [review]
ok let's do this

**Pull request merged**